src/target/llvm/llvm_instance.cc

```diff
@@ -297,9 +297,9 @@ llvm::TargetMachine* LLVMTargetInfo::GetOrCreateTargetMachine(bool allow_missing
         llvm_instance->createTargetMachine(triple_, cpu_, GetTargetFeatureString(), target_options_,
                                            reloc_model_, code_model_, opt_level_);
     target_machine_ = std::unique_ptr<llvm::TargetMachine>(tm);
-    if (!allow_missing) {
```

Comment:
This is the only place where `target_machine_` is assigned a value, so this check was actually sufficient.

Reply:
Emmmm but when `llvm::TargetRegistry::lookupTarget(triple_, error)` failed，the `target_machine_`  is also nullptr,

Reply:
Ah, you're right.